src/bundle/Security/Http/Authenticator/AuthenticatorDecorator.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Scheb\TwoFactorBundle\Security\Http\Authenticator;
+
+use Scheb\TwoFactorBundle\Security\Authentication\Token\TwoFactorTokenInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\AuthenticationContextFactoryInterface;
+use Scheb\TwoFactorBundle\Security\TwoFactor\Handler\AuthenticationHandlerInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\AnonymousToken;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authenticator\AuthenticatorInterface;
+use Symfony\Component\Security\Http\Authenticator\Passport\PassportInterface;
+
+class AuthenticatorDecorator implements AuthenticatorInterface
```

Comment:
Please implement `InteractiveAuthenticatorInterface` as well and then add:

```php
public function isInteractive(): bool
{
    return $this->decoratedAuthenticator instanceof InteractiveAuthenticatorInterface && $this->decoratedAuthentication->isInteractive();
}
```

That would make this authenticator fully compatible for all authenticators (I don't want to tell much about the `InteractiveAuthenticatorInterface`, as it's probably going to be deprecated/removed along with the `INTERACTIVE_LOGIN` event in an upcoming version)